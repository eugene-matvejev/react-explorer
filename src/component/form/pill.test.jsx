import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pill from './pill';

configure({ adapter: new Adapter() });

describe('<Pill/>', () => {
    const props = {
        label: '{{label}}',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Pill {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['data-cy', '{{data-cy}}'],
                ['data-id', '{{data-id}}'],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<Pill {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
