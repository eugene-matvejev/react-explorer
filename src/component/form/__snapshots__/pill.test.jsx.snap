// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill/> render with default/required props 1`] = `
<span
  className="pill"
  data-cy=""
>
  {{label}}
</span>
`;

exports[`<Pill/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<span
  className="pill"
  data-cy="{{data-cy}}"
>
  {{label}}
</span>
`;

exports[`<Pill/> render with optional props [::onClick] as "() => {}" 1`] = `
<span
  className="pill"
  data-cy=""
>
  {{label}}
  <button
    className="pill__control"
    data-cy="-remove"
    onClick={[Function]}
  >
    Ã—
  </button>
</span>
`;
